tests/src/test/fdc3.broadcast.ts

```diff
@@ -822,36 +1119,77 @@ export default () =>
       }
     }
 
-    function resetChannelsAppContext() {
-      channelsAppContext.broadcastMultipleItems = false;
-      channelsAppContext.useAppChannel = false;
-      channelsAppContext.contextBroadcasts.contact = false;
-      channelsAppContext.reverseMethodCallOrder = false;
-      channelsAppContext.broadcastExecutionComplete = false;
-    }
-
-    const executionCompleteListener = (
+    const waitForContext = (
       contextType: string,
+      testId: string,
       channel?: Channel
     ) => {
       return new Promise<Context>(async (resolve) => {
-        if (channel === undefined) {
-          const listener = await window.fdc3.addContextListener(
-            contextType,
-            (context) => {
+        let listener: Listener | null = null;
+        const handler = (context: AppControlContext) => {
+          if (testId) {
+            if (testId == context.testId) {
               resolve(context);
-              listener.unsubscribe();
+              if(listener) listener.unsubscribe();
+            } else {
+              console.warn(
+                `Ignoring ${contextType} context due to mismatched testId (expected: ${testId}, got ${context.testId})`
+              );
             }
-          );
-        } else {
-          const listener = await channel.addContextListener(
+          } else {
+            resolve(context);
+            if(listener) listener.unsubscribe();
+          }
+        };
+        if (channel === undefined) {
+          listener = await window.fdc3.addContextListener(
             contextType,
-            (context) => {
-              resolve(context);
-              listener.unsubscribe();
-            }
+            handler
           );
+        } else {
+          //App channels do not auto-broadcast current context when you start listening, so retrieve current context to avoid races
+          const currentContext  = await channel.getCurrentContext(contextType);
```

Comment:
This is causing issues. The problem with `await channel.getCurrentContext(contextType)` is that if it doesn't retrieve the context it will wait forever.

Reply:
It shouldn't, it should return `null`: https://fdc3.finos.org/docs/api/ref/Channel#getcurrentcontext

![image](https://user-images.githubusercontent.com/1701764/199489111-ae37abe9-4b1b-49b1-afa2-dfb4f3463eac.png)

Looks like there is a bug on our end when the context type is specified. This needs covering in a new test:
![image](https://user-images.githubusercontent.com/1701764/199489403-0744ec90-b135-4207-bb28-cf5ce5ebc234.png)

Reply:
For the time being I'll tweak this to not get stuck. But we'll probably need to add a test that checks behavior of `channel.getCurrentContext()` and `channel.getCurrentContext(type)` on new/unused channels.

Reply:
Take a look at the revised version, it shouldn't get stuck anymore at least.